packages/flame/lib/src/effects2/size_effect.dart

```diff
@@ -0,0 +1,46 @@
+import '../../components.dart';
+import '../../extensions.dart';
+import 'component_effect.dart';
+import 'effect_controller.dart';
+
+/// Change the size of a component over time.
```

Comment:
Will this work with a `ShapeComponent` too?

I wonder if `ComponentEffect<PositionComponent>` is the correct super-class even for this effect. One could easily imagine non-PositionComponent classes having the notion of size. Say, a `Viewport`?

Reply:
It will work on the `ShapeComponent`, since it is a `PositionComponent`. Since right now the `size` is defined on `PositionComponent` I think it will be enough for now, we can evolve that later if we would see the need to have these effects on other things, like the viewport.